### PR TITLE
Add openapi documentation

### PR DESCRIPTION
**Pull request by e00E**
Part of #1264 .

### Test Plan
Load the file in https://editor.swagger.io/ and see the api docs.

### CODE REVIEW

File: price-estimator/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      example: 1-7
+  UnitParameter:
+    name: unit
+    in: query
+    description: If `baseunits` (the default) all amounts are denominated in the "natural" unit of the respective token given by the number of decimals specified through the ERC20 interface. If set to `atoms` all amounts are denominated in the smallest available unit (atom) of the token.
+    required: false
+    schema:
+      type: string
+      enum: [baseunits, atoms]
+      example: baseunits
```
Comment (e00E) [inline price-estimator/openapi.yml:63]: I am assuming Nick's PR that changes the unit parameter to work like this instead of `atoms=true|false` gets merged. That PR deprecates the atoms syntax so we shouldn't document it.

File: price-estimator/src/filter.rs [hunk 2]
```diff
@@ -40,13 +39,12 @@ pub fn all(
...
 
-async fn handle_rejection(err: Rejection) -> Result<impl Reply, Infallible> {
+async fn handle_rejection(err: Rejection) -> Result<impl Reply, Rejection> {
     let code;
     let message;
 
     if err.is_not_found() {
-        code = StatusCode::NOT_FOUND;
-        message = "invalid url path";
+        return Err(reject::not_found());
```
Comment (e00E) [inline price-estimator/src/filter.rs:47]: The reason for this is that rejections should be used to signal that a filter cannot handle a request but another filter might. So we shouldn't return a response in the case of the path not matching so that the next filter can try to match the path.

File: price-estimator/src/openapi.rs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+use warp::{Filter, Rejection, Reply};
+
+pub fn openapi() -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    warp::path("openapi").and(warp::fs::dir("price-estimator/openapi"))
```
Comment (e00E) [inline price-estimator/src/openapi.rs:4]: Will make the path on disk configurable in another PR.

File: price-estimator/openapi.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      type: string
+      example: 1-7
+  UnitParameter:
+    name: unit
+    in: query
+    description: If `baseunits` (the default) all amounts are denominated in the "natural" unit of the respective token given by the number of decimals specified through the ERC20 interface. If set to `atoms` all amounts are denominated in the smallest available unit (atom) of the token.
+    required: false
+    schema:
+      type: string
+      enum: [baseunits, atoms]
```
Comment (nlordell) [inline price-estimator/openapi.yml:92]: Should you add `default: baseunits`?

File: price-estimator/swagger-ui.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
```
Comment (nlordell) [inline price-estimator/swagger-ui.sh:1]: Personally, I'm kind of against having `price-estimator` host the swagger UI HTML. Its fairly easy to go to https://petstore.swagger.io/ and just input `http://localhost:8080/openapi/openapi.yaml` to get the UI. I think just having the OpenAPI definition is enough.

If we really want to include host the swagger UI, I think it should be added in a separate PR.

Comment (e00E) [reply price-estimator/swagger-ui.sh:1]: Ah so we would only host the one file but not the full ui. I'll have to check if this will work with CORS (setting allow origin header).
I didn't think of this possibility when I made the PR but I do like self hosting all the required files because it works without any external website. Why are you against this?

Comment (nlordell) [reply price-estimator/swagger-ui.sh:1]: > I'll have to check if this will work with CORS

Good point. We may need to enable CORS correctly.

> Why are you against this?

Not against self-hosting, I just don't think it should be the `price-estimator` that hosts this, and maybe this should be hosted in the dev portal instead of the service. Another concern of mine is that the zip bundled UI may contain Swagger branding that we maybe aren't allowed to distribute.

Another thing to consider is using `swagger-ui(-dist)?` NPM package and host that in a little HTML page of our own instead of downloading from a ZIP bundle.

**Review by fleupold — APPROVED** [review]
The creation script wasn't fully working for me (comments inline). Agree that we should maybe host it somewhere in a separate container (or for now just link the swagger hosted link in the dev portal).

**Comment by e00E** [comment]
Changed PR to only include the yaml file on disk. No http route for it. Then we can decide the rest in another PR.

File: price-estimator/openapi.yml [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    Unit:
+      name: unit
+      in: query
+      description: If `baseunits` (the default) all amounts are denominated in the "natural" unit of the respective token given by the number of decimals specified through the ERC20 interface. If set to `atoms` all amounts are denominated in the smallest available unit (atom) of the token.
+      required: false
+      schema:
+        type: string
+        enum: [baseunits, atoms]
+        default: baseunits
+        example: baseunits
```
Comment (nlordell) [inline price-estimator/openapi.yml:139]: I think the `batchId` query parameter is missing.

**Review by nlordell — APPROVED** [review]
Thanks. I think I agree with @fleupold that we can have a `Dockerfile` to build an image for hosting the Swagger UI.
